### PR TITLE
feat: add support for breakout rooms

### PR DESCRIPTION
**Pull request by marcos-cereijo-pexip**
Change also the message overlay in all the breakout rooms.

### CODE REVIEW

**Comment by marcos-cereijo-pexip** [comment]
Adding a selector to change between:

- All rooms.
- Main room.
- Breakout room 1
- Etc.

File: src/index.ts [hunk 1]
```diff
@@ -1,23 +1,47 @@
...
+setPlugin(plugin)
+
 let button: Button<'settingsMenu'> | null = null
 let creatingButton = false
 
+const breakoutRooms = getBreakoutRooms()
+
+plugin.events.authenticatedWithConference.add(async () => {
+  breakoutRooms.clear()
+})
```
Comment (TsvetozarTsonev) [inline src/index.ts:22]: what happens if we don't have this 😄

File: src/index.ts [hunk 2]
```diff
@@ -1,23 +1,47 @@
...
+  breakoutRooms.set(breakoutRoom.breakout_uuid, '')
+})
+
+plugin.events.breakoutEnd.add(async (breakoutRoom) => {
+  breakoutRooms.delete(breakoutRoom.breakout_uuid)
+})
+
 plugin.events.conferenceStatus.add(async ({ id, status }) => {
-  if (id === 'main' && !status.directMedia && status.started) {
+  if (breakoutRooms.has(id) && status.breakoutName != null) {
```
Comment (TsvetozarTsonev) [inline src/index.ts:33]: do we need this extra `null` check?

**Review by TsvetozarTsonev — COMMENTED** [review]
Nice!

A couple more comments:
- you can get rid of the `breakoutRooms.ts` file and you can just hold them in the `index` file and pass them as a prop to `handleSubmitMessage` as they are only used there
- for the `plugin.ts`, you can just `export let plugin` and assign it in the index, and avoid the getters and setters, usually its a good practice but its a bit of an overkill here (IMHO)

File: src/index.ts (hunk 1 above)
Comment (marcos-cereijo-pexip) [reply src/index.ts:22]: If you don't do this, leave the room and connect again, you will see the old breakout rooms when you try to send a new message overlay.

**Comment by marcos-cereijo-pexip** [comment]
> Nice!
> 
> A couple more comments:
> 
> * you can get rid of the `breakoutRooms.ts` file and you can just hold them in the `index` file and pass them as a prop to `handleSubmitMessage` as they are only used there
> * for the `plugin.ts`, you can just `export let plugin` and assign it in the index, and avoid the getters and setters, usually its a good practice but its a bit of an overkill here (IMHO)

@TsvetozarTsonev , I changed the part related to the `breakoutRooms`, but I cannot do the same with the plugin (or I don't know exactly how to do it). If I import the `index.ts` in another file, it's triggered twice.

**Comment by marcos-cereijo-pexip** [comment]
@TsvetozarTsonev I modified the code and  now I export the `plugin` variable directly.

**Review by TsvetozarTsonev — APPROVED** [review]

**Pull request merged**